app/Http/Controllers/Web/GroupTypesController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Rogue\Http\Controllers\Web;
+
+use Rogue\Models\GroupType;
+use Illuminate\Http\Request;
+use Rogue\Http\Controllers\Controller;
+
+class GroupTypesController extends Controller
+{
+    /**
+     * Create a controller instance.
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $this->middleware('role:admin');
+
+        $this->rules = [
+            'name' => 'required|unique:group_types',
+        ];
+    }
+
+    /**
+     * Create a new group type.
+     */
+    public function create()
+    {
+        return view('group-types.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     */
+    public function store(Request $request)
+    {
+        $request->validate($this->rules);
+
+        $groupType = GroupType::create($request->all());
+
+        // Log that a group type was created.
+        info('group_type', ['id' => $groupType->id]);
+
+        return redirect('group-types/' . $groupType->id);
+    }
+
+    /**
+     * Edit an existing action.
+     *
+     * @param  \Rogue\Models\GroupType  $groupType
+     */
+    public function edit(GroupType $groupType)
+    {
+        return view('group-types.edit')->with([
+            'groupType' => $groupType,
+        ]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Rogue\Models\GroupType  $groupType
+     * @param  \Illuminate\Http\Request  $request
+     */
+    public function update(GroupType $groupType, Request $request)
+    {
+        $this->validate($request, $this->rules);
```

Comment:
Let's add an override to the [`unique`](https://laravel.com/docs/5.5/validation#rule-unique) rule here to exclude this resource from the uniqueness constraint (since otherwise we'd have to change the name every time we hit this form since it'd conflict with itself).

(I know this doesn't really break things now since this is the only field, but this could make for a confusing surprise if we were to ever add any other fields to this form!)

Reply:
Ah - is what's going on over in the `CampaignsController` what you're referring to?

https://github.com/DoSomething/rogue/blob/d85d8429316aabb3834a12e6a504e81c3d252bbd/app/Http/Controllers/Web/CampaignsController.php#L80-L82

I'm still wrapping my head around validation in Laravel... ideally once we start making Groups editable (added in #1031) - we want a unique index on `group_type_id` and `name`, and been wondering about how to handle that in the form (vs just relying on the scary server error error broken screen)

Reply:
I'm going to merge this for now, since approved and can fix this up in another PR -- there's some more on deck.